### PR TITLE
[Xamarin.Android.Build.Tasks] Always create the debug.keystore

### PR DESCRIPTION
**Pull request by dellis1972**
Fixes https://bugzilla.xamarin.com/show_bug.cgi?id=61065

If a user sets the `$(AndroidKeyStore)` to `True` for a
debug configuration they will get the following error.

	[MESSAGE] jarsigner error: java.lang.RuntimeException: keystore load: /Users/$(user)/.local/share/Xamarin/Mono for Android/debug.keystore (No such file or directory)

Note this will only happen if they have NEVER build with
`$(AndroidKeyStore)` set to `False`. This is because we
don't always generate the `debug.keystore`.

So rather than hitting this issue we should always generate the
`debug.keystore` regardless of what the setting for `$(AndroidKeyStore)`
is. This commit splits out the keystore generation into its own
target. This target will check the existance of the file and
generate it if required. We also move all the debug key property
values into Properties so that they can be updated in one place.

### CODE REVIEW

**Comment by jonpryor** [comment]
This PR confuses me.

The original bug is that the `Mono.Android.DebugRuntime*.apk` shared runtime is signed with the "debug key," which isn't always created. If it hasn't been created, then package installation fails.

*Additionally*, the shared runtime creation logic within `androidtools` *hardcodes* the values used here, e.g. `androiddebugkey` as the `jarsigner` key alias.

Using `$(_ApkDebugKeyAlias)` to store `androiddebugkey` and then allowing `$(_ApkDebugKeyAlias)` to be overridden -- as the `//CreateProperty/Output[@PropertyName='_ApkDebugKeyAlias']/@Condition` supports -- is a recipe for future errors, as some enterprising developer decides to override `$(_ApkDebugKeyAlias)` within their `.csproj` and then wonders why things break in wonderful and mysterious ways.

I would prefer that the `<AndroidCreateDebugKey/>` invocation just hardcoded *all* the values. It's the "debug" key; we control it.

File: src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets [hunk 1]
```diff
@@ -2454,9 +2472,23 @@ because xbuild doesn't support framework reference assemblies.
 
   <Delete Files="$(_UploadFlagFile)" Condition="Exists ('$(_UploadFlagFile)')" />
 </Target>
-  
 
-<Target Name="_ResolveAndroidSigningKey" DependsOnTargets="$(_OnResolveMonoAndroidSdks)">
+<Target Name="_CreateAndroidDebugSigningKey"
+		Condition="!Exists ('$(_ApkDebugKeyStore)')"
+		DependsOnTargets="$(_OnResolveMonoAndroidSdks)"
```
Comment (jonpryor) [inline src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets]: Why is this `DependsOnTargets` required? The only caller -- `_ResolveAndroidSigningKey` -- has `$(_OnResolveMonoAndroidSdks)` within its `DependsOnTargets` anyway.

**Comment by dellis1972** [comment]
@jonpryor I removed the properties.. One thing to note.. our assumption that `CreateProperty` does NOT override command line arguments is find for msbuild but NOT for xbuild. It just happily overwrites the values. So I have had to leave the condition on the `_ApkDebugKeyStore` CreateProperty call.

**Comment by EmilAlipiev** [comment]
Hi, interestingly it also fails if i refer to another keystore file. VS still looking for debug.keystore. please see the forum [post ](https://forums.xamarin.com/discussion/comment/399430)more information.

**Pull request merged**